storage/local/chunk.go

```diff
@@ -188,23 +188,27 @@ type chunk interface {
 	add(sample *metric.SamplePair) []chunk
 	clone() chunk
 	firstTime() clientmodel.Timestamp
-	lastTime() clientmodel.Timestamp
 	newIterator() chunkIterator
 	marshal(io.Writer) error
 	unmarshal(io.Reader) error
 	unmarshalFromBuf([]byte)
 	encoding() chunkEncoding
-	// values returns a channel, from which all sample values in the chunk
-	// can be received in order. The channel is closed after the last
-	// one. It is generally not safe to mutate the chunk while the channel
-	// is still open.
-	values() <-chan *metric.SamplePair
 }
 
 // A chunkIterator enables efficient access to the content of a chunk. It is
 // generally not safe to use a chunkIterator concurrently with or after chunk
 // mutation.
 type chunkIterator interface {
+	// length returns the number of samples in the chunk.
+	length() int
+	// Gets the timestamp of the n-th sample in the chunk.
+	getTimestampAtIndex(int) clientmodel.Timestamp
```

Comment:
Do we have to prefix `get` here? It's not `getLength()` and `getValues()` either.

As it's purely internal it doesn't matter too much.